### PR TITLE
homework 10

### PR DESCRIPTION
**Pull request by artsiom-tsaryonau**

### CODE REVIEW

File: homework10/src/main/java/com/epam/lab/mentoring/ClassicModelExample.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        });
+
+        try {
+            Thread.sleep(4000); // waiting for threads to finish execution just in case
+            System.out.println("Calculation result => " + output);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
```
Comment (AShynkevich) [inline homework10/src/main/java/com/epam/lab/mentoring/ClassicModelExample.java]: Remove the line

File: homework10/src/main/java/com/epam/lab/mentoring/Main.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.epam.lab.mentoring;
+
+public class Main {
+
+    public static void main(String[] args) {
+        new ClassicModelExample().calculate();
+
+        new UtilConcurrentExample().calculate();
+    }
+
```
Comment (AShynkevich) [inline homework10/src/main/java/com/epam/lab/mentoring/Main.java]: Remove the line

File: homework10/src/main/java/com/epam/lab/mentoring/UtilConcurrentExample.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class UtilConcurrentExample extends AbstractExample {
+    @Override
+    public void calculate() {
+        ExecutorService executorService = Executors.newFixedThreadPool(getInputSize());
+        IntStream.range(0, getInputSize()).forEach(idx -> {
+            executorService.submit(() -> this.calculate(idx));
+        });
+        executorService.shutdown();
+
+        while (!executorService.isTerminated()) {}
```
Comment (AShynkevich) [inline homework10/src/main/java/com/epam/lab/mentoring/UtilConcurrentExample.java]: It looks weird. May executor has method which doesn't need loop to wait

Comment (artsiom-tsaryonau) [reply homework10/src/main/java/com/epam/lab/mentoring/UtilConcurrentExample.java]: updated

**Pull request merged**